chia/_tests/timelord/test_new_peak.py

```diff
@@ -348,6 +348,64 @@ async def test_timelord_new_peak_unfinished_eos(
                     == new_peak.reward_chain_block.get_hash()
                 )
 
+    @pytest.mark.anyio
+    async def test_timelord_new_peak_node_sync(
+        self,
+        one_node: tuple[list[FullNodeService], list[FullNodeSimulator], BlockTools],
+        timelord: tuple[TimelordAPI, ChiaServer],
+        default_1000_blocks: list[FullBlock],
+    ) -> None:
+        [full_node_service], _, bt = one_node
+        full_node = full_node_service._node
+        async with create_blockchain(bt.constants, 2) as (b1, _):
+            async with create_blockchain(bt.constants, 2) as (b2, _):
+                timelord_api, _ = timelord
+                for block in default_1000_blocks:
+                    await _validate_and_add_block(b1, block)
+                    await _validate_and_add_block(b2, block)
+                    await full_node.add_block(block)
+
+                peak = timelord_peak_from_block(b1, default_1000_blocks[-1])
+                assert peak is not None
+                assert timelord_api.timelord.new_peak is None
+                await timelord_api.new_peak_timelord(peak)
+                assert timelord_api.timelord.new_peak is not None
+                assert (
+                    timelord_api.timelord.new_peak.reward_chain_block.get_hash() == peak.reward_chain_block.get_hash()
+                )
+                await time_out_assert(60, peak_new_peak_is_none, True, timelord_api)
+                # make two new blocks on tip, block_2 has higher total iterations
+                block_1 = bt.get_consecutive_blocks(1, default_1000_blocks)[-1]
+                block_2 = bt.get_consecutive_blocks(
+                    1, default_1000_blocks, min_signage_point=block_1.reward_chain_block.signage_point_index
+                )[-1]
+                assert block_2.weight == block_1.weight
```

Comment:
it would probably be good to also cover the case where the new peak has a lower weight

Reply:
test_timelord_new_peak_basic in mind covers that case

Reply:
theres a basic test that checks lower wight test_timelord_new_peak_basic